l1-contracts/src/core/messagebridge/MessageBox.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 Aztec Labs.
+pragma solidity >=0.8.18;
+
+import {IRegistryReader} from "./IRegistryReader.sol";
+
+/**
+ * @title MessageBox
+ * @author Aztec Labs
+ * @notice Data structure used in both Inbox and Outbox for keeping track of entries
+ * Implements a multi-set storing the multiplicity (count for easy reading) at the entry.
+ */
+abstract contract MessageBox {
+  error MessageBox__Unauthorized();
+  error MessageBox__NothingToConsume(bytes32 entryKey);
+  error MessageBox__OversizedContent();
+
+  /// @dev Actor on L1. ChainID if multiple L1s tap into the same Aztec instance.
+  struct L1Actor {
```

Comment:
Would be fine to add a note on the structures here that they won't be used in storage only in memory so packing is not done.

Reply:
good idea